openc3-cosmos-cmd-tlm-api/app/controllers/storage_controller.rb

```diff
@@ -111,10 +111,23 @@ def exists
 
   def download_file
     return unless authorization('system')
-    volume = ENV[params[:volume]] # Get the actual volume name
-    raise "Unknown volume #{params[:volume]}" unless volume
-    filename = "/#{volume}/#{params[:object_id]}"
-    filename = sanitize_path(filename)
+    if params[:volume]
+      volume = ENV[params[:volume]] # Get the actual volume name
+      raise "Unknown volume #{params[:volume]}" unless volume
+      filename = "/#{volume}/#{params[:object_id]}"
+      filename = sanitize_path(filename)
+    elsif params[:bucket]
```

Comment:
Why no bucket before?

Reply:
We never did a download from a bucket before ... just volumes. We always used the `get_download_presigned_request` instead because the result was always a file. Now we're downloading to memory so we can display.